### DISCUSSION TITLE
Having trouble generating a manifest.json

### DISCUSSION THREAD

**Discussion by rob-king-volpara** [microsoft/aspire#4578] [Q&A] [answered]
I have a simple Aspire stack - a host app, a worker app, a storage resource with a queue and blobs. It runs absolutely fine, but when I try to create the manifest.json file using the command:

```
dotnet run --publisher=manifest --output-path=manifest.json
```

It fails trying to output whatever comes immediately after the storage resource with:

`System.InvalidOperationException: The endpoint 'queue' is not allocated for the resource 'Storage'

For the reference, the program.cs starts like this:

```
IDistributedApplicationBuilder builder = DistributedApplication.CreateBuilder(args);

IResourceBuilder<AzureStorageResource> storage = builder.AddAzureStorage("Storage")
    .RunAsEmulator()
    .WithAnnotation(new ContainerImageAnnotation
    {
        Registry = "mcr.microsoft.com",
        Image = "azure-storage/azurite",
        Tag = "3.30.0"
    });

IResourceBuilder<AzureQueueStorageResource> queues = storage.AddQueues("Queues");
IResourceBuilder<AzureBlobStorageResource> blobs = storage.AddBlobs("Blobs");
```

The Storage.module.bicep file looks to generate ok, but the manifest.json just ends at the failure point:

```
{
  "resources": {
    "Storage": {
      "type": "azure.bicep.v0",
      "path": "Storage.module.bicep",
      "params": {
        "principalId": "",
        "principalType": ""
      }
    },
    "Queues": {
      "type": "value.v0"
```

I'm running the GA bits too.

**Comment by davidfowl** [comment]
It’s because of the custom code you wrote to add the annotation. Use the callback overload in RunAsEmulator to change the image like I showed.

**Reply by davidfowl** [reply, marked as answer]
```
IResourceBuilder<AzureStorageResource> storage = builder.AddAzureStorage("Storage")
    .RunAsEmulator(c => c.WithImageTag("3.30.0"));
```

**Reply by rob-king-volpara** [reply]
Bingo, that was it.

Legend.
